### DISCUSSION TITLE
Add the ability to stack skins like Minecraft resource packs. (basically automatic skin mixing)

### DISCUSSION THREAD

**Discussion by OwenCMYK** [ppy/osu#27378] [Ideas] [answered]
The idea is rather simple, but probably horribly complex in practice: Give players an option to add skins on top of their current skin to tweak certain elements. For example: A skin might only skin hitsounds and make "great" judgements transparent, so stacking it on top of your main skin would turn your main skin into a "pro" skin. Skins could be stacked with a priority order, so if there are duplicate elements (elements skinned by more than one of your skins), then whichever one has higher priority would win out. For the skin editor, you would probably have to select which skin you want to edit (for the sake of not causing enormous issues).

**Comment by trustytrojan** [comment]
This is completely unnecessary as you're better off just modifying your skin by exporting it to a `.osk` file, extracting it to a folder, adding the elements of other skins into it, re-zipping it into a `.osk` file and re-importing it. Or you can wait until the skin editor is fully implemented.

**Reply by OwenCMYK** [reply]
I know how to modify skins, I've done it myself. But there are a lot of cases where this system would be much cleaner, especially if you didn't want to make a new skin for every possible combination of elements you might want. Hence why this feature exists in some other games, and it's usefulness in those games speaks for itself.

The feature might be too much work to be worth implementing, but I highly doubt it would be useless if it were implemented.

**Comment by Gabixel** [comment]
I imagine there are many implications to get to this functionality, but I have this vague memory from several years ago of peppy briefly discussing this on Discord and... I could be wrong... but I think that's (roughly) actually one of their goals.
Probably best to get a direct answer from him at this point.

**Reply by Gabixel** [reply, marked as answer]
Huh, looks like I found that conversation. I don't know if it has evolved as a thing and/or if this work that @peppy was doing still exists.

![image](https://github.com/ppy/osu/assets/43073074/c838b9a9-4aa1-46fe-ba51-2944ae7a7485)

Here's the discussion video:

https://github.com/ppy/osu/assets/43073074/e3c8b596-9d62-4d52-9fb6-bbe42790ce45

**Reply by OwenCMYK** [reply]
YES that is literally exactly what I was thinking in every way. Thank you, I'm glad to know they've considered it

**Reply by Gabixel** [reply]
I'd be curious if we could keep this discussion open to keep track of it (since it's planned _in some way_, not sure if in the settings or the skin editor), and I don't think there have ever been any specific discussions opened about that.
I also found this other discussion that may have something to do with this: https://github.com/ppy/osu/discussions/18314

**Reply by OwenCMYK** [reply]
Yeah, I this feature would naturally solve that problem as a side effect. Since you could just make skins that are hitsound packs (which I imagine many people will do). It might also be cleaner to separate Argon Pro into an addon pack for Argon, so then people will have a "pro" addon pack that comes with the game that they can apply to any skin they choose. (like triangles or circles for example)
